sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py

```diff
@@ -71,28 +76,112 @@ def _need_new_token(self):
         return not self._token or self._token.expires_on - time.time() < 300
 
 
-class BearerTokenCredentialPolicy(_BearerTokenCredentialPolicyBase, SansIOHTTPPolicy):
+class BearerTokenCredentialPolicy(_BearerTokenCredentialPolicyBase, HTTPPolicy):
     """Adds a bearer token Authorization header to requests.
 
-    :param credential: The credential.
-    :type credential: ~azure.core.TokenCredential
-    :param str scopes: Lets you specify the type of access needed.
-    :raises: :class:`~azure.core.exceptions.ServiceRequestError`
+    :param ~azure.core.credentials.TokenCredential credential: credential for authorizing requests
+    :param str scopes: required authentication scopes
     """
 
     def on_request(self, request):
-        # type: (PipelineRequest) -> None
+        """This method is for backward compatibility. It has no implementation."""
```

Comment:
It's not backward compatible, since if I inherit and call `super().on_request()` is suddently does nothing.

Reply:
This class didn't override the method, so before this change you would have been calling SansIOHTTPPolicy.on_request [which also does nothing](https://github.com/Azure/azure-sdk-for-python/blob/9dec418ad621ac75f217e56e901f15b6624800b0/sdk/core/azure-core/azure/core/pipeline/policies/_base.py#L92).